syntax = "proto3";

package myrtlespeech.protos;

import "myrtlespeech/protos/stage.proto";


// Preprocessing steps to apply to the input data.
message PreProcessStep {
  Stage stage = 1;

  oneof pre_process_step {
    MFCC mfcc = 2;
    Standardize standardize = 3;
    ContextFrames context_frames = 4;
  }
}

// Computes Mel-frequency cepstral coefficients (MFCC).
message MFCC {
  // Number of cepstrum.
  uint32 numcep = 1;

  // Length of the analysis window in seconds.
  float winlen = 2;

  // Step between successive windows in seconds.
  float winstep = 3;
}


// Standardize the input.
message Standardize {
}


message ContextFrames {
  // Number of context frames (channels) to add.
  uint32 n_context = 1;
}
